src/index.js

```diff
@@ -0,0 +1,54 @@
+import './scss/style.scss';
+import searchIcon from './images/search-icon.png';
+
+let weather = {
+  apiKey: 'c20d664fe15511ff6b176f9e20df512f',
+  fetchweather: function (city) {
+    fetch(
+      'http://api.openweathermap.org/data/2.5/weather?q= ' +
+        city +
+        '&units=metric&appid=' +
+        this.apiKey,
+    )
+      .then((response) => response.json())
+      .then((data) => this.displayweather(data));
+  },
+  displayweather: function (data) {
+    const { name } = data;
+    const { description } = data.weather[0];
+    const { temp, humidity } = data.main;
+    const { speed } = data.wind;
+
+    document.querySelector('.city').innerText = 'Weather in ' + name;
+    document.querySelector('.icon').src =
+      'https://openweathermap.org/img/wn/"+ icon +" 01n@.png';
+    document.querySelector('.description').innerText = description;
+    document.querySelector('.temp').innerText = temp + '°C';
+    document.querySelector('.humidity').innerText =
+      'Humidity: ' + humidity + '%  ';
+    document.querySelector('.wind').innerText =
+      'Wind speed ' + speed + ' km/h  ';
+    document.querySelector('.weather').classList.remove('Loading');
+    document.body.style.backgroundImage =
+      "url('https://source.unsplash.com/1600x900/?" + name + " ')";
+  },
+  search: function () {
+    this.fetchweather(document.querySelector('.search-bar').value);
+  },
+};
+
+document.querySelector('.search button').addEventListener('click', function () {
+  weather.search();
+});
+
+document
+  .querySelector('.search-bar')
+  .addEventListener('keyup', function (event) {
+    if (event.key == 'Enter') {
+      weather.search();
+    }
+  });
+
+weather.fetchweather();
```

Comment:
The error in the console, is caused by this line, it is expecting an argument `city`, so since none is passed when it application first run, the city value would be undefined hence the error. To address this pass a default value to the `fetchweather()` in the function definition above, that way when no argument is passed the default value would be used.